js/main.js

```diff
@@ -1,107 +1,152 @@
-const addBtn = document.querySelector('.add-btn');
-addBtn.onclick = addBook;
-
-let bookCollection = [];
 let messageOn = false;
-
-// Helper functions
-function renderEmptyMessage() {
-  messageOn = true;
-  const bookList = document.querySelector('.books-display');
-  bookList.innerHTML = '';
-  const message = document.createElement('span');
-  message.innerText = 'Your book collection is empty. Kindly add your favorite books.';
-
-  bookList.appendChild(message);
-  return bookCollection;
-}
-
-function isCollectionEmpty() {
-  if (bookCollection.length === 0) renderEmptyMessage();
+class BooksCollection {
+  constructor() {
+    return [];
+  }
+
+  static isCollectionEmpty() {
+    if (this.length === 0) {
+      this.renderEmptyMessage();
+    }
+  }
+
+  static saveCollection(collection) {
+    localStorage.setItem('books', JSON.stringify(collection));
+  }
 }
 
-// Model
-const savedCollection = JSON.parse(localStorage.getItem('books'));
-
-bookCollection = savedCollection ?? renderEmptyMessage();
-isCollectionEmpty();
-
-if (!messageOn) renderBooks(bookCollection);
-
-// Saves Collection
-function saveCollection() {
-  localStorage.setItem('books', JSON.stringify(bookCollection));
-}
+let bookCollection = new BooksCollection();
 
-// Creates a Book
-function CreateBook(id, title, author) {
-  this.id = id;
-  this.title = title;
-  this.author = author;
+// Controller
+class CreateBook {
+  constructor(id, title, author) {
+    this.id = id;
+    this.title = title;
+    this.author = author;
+  }
```

Comment:
- [ ] Please add the functionality of creating (add) and removing a book as a method to the `BookCollection` class as mentioned in the project requirements.

![image](https://user-images.githubusercontent.com/68796232/182392242-2d04ade4-5019-40c2-a86f-4ff9a8812952.png)

Reply:
@Omar-Muhamad thank you for your review, we have made a class called Class actions that contains the methods you mentioned [Here are those lines](https://github.com/michael-duke/Awesome-Books/blob/plain-JS-with-classes/js/main.js#L76-L95) 😄

Reply:
We also call those functions [here for removing a Book](https://github.com/michael-duke/Awesome-Books/blob/plain-JS-with-classes/js/main.js#L123) and also [here for adding a Book](https://github.com/michael-duke/Awesome-Books/blob/plain-JS-with-classes/js/main.js#L152) Kindly take note @Omar-Muhamad

Reply:
Hey Michael, I know you have the methods I just need you to move them to the BookCollection class as mentioned in the comment above. The reason for this is that you must have all the functionalities of the class inside the class. not in a separate class. that's what classes are for, containing the functionality and state of a single entity.

Reply:
Okay noted @Omar-Muhamad but we made it so because it is a book action the requirement is to use class methods as you have mentioned. In our implementation we use the BookCollection instance in our BookAction class
We will convey and try what we can do on our end, thank you for your feedback